### ISSUE TITLE
Set-JiraIssueLabel function merges labels into one

### ISSUE THREAD

**Issue by anthonyjvoss** [AtlassianPS/JiraPS#82]
In the past I was able to use this function to add a single new label to a Jira issue.  Since updating the module, I'm now having the issue where the new label is appended to the existing label or labels.

i.e. If a Jira issue already had the label 'test', and I wanted to add the label 'dev', it would instead create a new label 'testdev' instead of 2 separate labels.

This seems to occur on Line 115 of the script function for Set-JiraIssueLabel where the label(s) to add are combined together:

$newLabels = $currentLabels + $Add | Where-Object -FilterScript {$_ -ne $null -and $Remove -notcontains `$_}

**Comment by anthonyjvoss** [comment]
I found a workaround by setting the current labels variable as an empty array and then adding the Jira issue labels property to this array:

$currentLabels = @()
$currentLabels += $issueObj.labels

This allows for separate labels to be added without combining them into one long string value.

Notes: Jira v7.0

**Comment by lipkau** [comment]
how are you using the function?
I can't find anything that could explain a wrong behavior in the code

```
# add two new labels to an issue:
Set-JiraIssueLabel -Add "foo","bar"

# replace all labels. remove all and add two new labels to an issue:
Set-JiraIssueLabel -Set "foo","bar"
```

**important to note: if you omit `-Add`, `-Set` will be used by default (as the name of the cmdlet suggests)**

**Comment by kittholland** [comment]
$issueObj = Get-JiraIssue -InputObject $i -Credential $Credential
                $currentLabels = $issueObj.labels

If issueObj.labels is a string, then calling the addition method will concat to the existing string rather than adding a new item:

                    $newLabels = $currentLabels + $Add | Where-Object -FilterScript {$_ -ne $null -and $Remove -notcontains $_}

If issueObj.labels is already an array, it works fine.

**Comment by anthonyjvoss** [comment]
@lipkau - sorry for not getting back to you.  Attached is the script I created to get the Jira issue, and then set the Jira issue label.  
@kittholland  - you are correct, and this is why I had to put my workaround in place to ensure an array was created instead of passing string values.  This is what I was originally curious about, to determine if any changes were made to this module.  This used to not occur when adding an additional label to an issue, but started occurring in this newer version of PSJira.

[Add-JiraLabel.txt](https://github.com/replicaJunction/PSJira/files/937258/Add-JiraLabel.txt)
